src/components/textFields/AdvancedOpenTextField.js

```diff
@@ -1,239 +1,272 @@
-    }
-
-    var renderComponent = true;
-    if (data.questionDependency != undefined && props.response != null) {
-        renderComponent = dependencyParser(props.response, data, dependencyID)
-    }
-
-    if (renderComponent){
-        return(
-            <QuestionSection key={key}
-                title={data.question}
-                helperText={data.helperText}
-                errorMessage={ErrorMsg()}
-                tooltip={tooltip()}
-                required={data.required}
-            >
-                <HStack>
-                    <Input
-                        placeholder='Place your Text'
-                        value = {valueSet(currId)}
-                        onChangeText={(text) => onTextChange(text)}
-                    />
-                    {RenderHeaderIcon()}
-                </HStack>
-            </QuestionSection>
-        )
-    }
-    else {
-        return null;
-    }
+	const [visible, setVisible] = React.useState(false);
+	const [value, setValue] = React.useState("");
+	const [buttonAppearance, setButtonAppearance] = React.useState("outline");
+	const [invalidLength, setInvalidLength] = React.useState(false);
+	const [invalidVin, setInvalidVin] = React.useState(false);
+	const [isInvalid, setIsInvalid] = React.useState(false);
+	const {
+		data,
+		key,
+		id,
+		questionReducer,
+		submitFunction,
+		pageChange,
+		importFrom,
+		updateRoad,
+		dependencyID,
+	} = props;
+	let currId = data.humanReadableId;
+	let status;
+	const reducerData = questionReducer.data.find((entry) => entry.id == id);
+	let existingData = !reducerData?.response ? null : reducerData.response;
+
+	// Populate if value already exists in redux
+	if (!value) {
+		if (existingData != null) {
+			if (
+				existingData[currId] != null &&
+				existingData[currId] != "" &&
+				!value
+			) {
+				setValue(existingData[currId]);
+			}
+		}
+	}
+
+	const submitField = (text) => {
+		//submits data to reducer and is shown as complete
+		submitFunction({ id, question: currId, selection: text });
+		switch (currId) {
+			case Constants.VIN_ID: // VIN question id
+				if (!vinValidator.validate(value)) {
+					setInvalidVin(true);
+				} else {
+					setInvalidVin(false);
+				}
+				break;
+			default:
+				break;
+		}
+		ErrorMsg();
+		setButtonAppearance("filled");
+	};
+
+	const onTextChange = (text) => {
+		//Updated the reducer when user types
+		let localText = text;
+		if (!text) {
+			submitFunction({ id, question: currId, selection: null });
+			setIsInvalid(false);
+		}
+		let textFieldValidation = TextFieldValidation;
+		textFieldValidation.submitField(localText);
+		let localStatus = textFieldValidation.status;
+		if (localStatus) {
+			setIsInvalid(false);
+		} else {
+			setIsInvalid(true);
+		}
+		switch (currId) {
+			case Constants.VIN_ID: // VIN question id
+				if (!vinValidator.validate(value)) {
+					setInvalidVin(true);
+				} else {
+					setInvalidVin(false);
+				}
+				break;
+			default:
+				break;
+		}
```

Comment:
this block could use some more comments - what does the local status mean? why is this only checking for vin validation as opposed to other validation / how could it be extended for other validation?